server/index.js

```diff
@@ -141,6 +142,81 @@ app.get('/api/comments/:postId', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.post('/api/likes', (req, res, next) => {
+  const { postId, userId } = req.body;
+  if (!postId || !userId) {
+    throw new ClientError(400, 'postId and userId are required fields');
+  }
+  const sql = `
+    insert into "likePosts" ("postId", "userId")
+    values ($1, $2)
+    on conflict do nothing
+    returning *
+  `;
+
+  const params = [postId, userId];
+
+  db.query(sql, params)
+    .then(result => {
+      const [newLike] = result.rows;
+      res.status(201).json(newLike);
+    })
+    .catch(err => next(err));
+});
+
+app.get('/api/likes/:postId', (req, res, next) => {
+  const postId = Number(req.params.postId);
+  const firstSql = `
+    select count("l".*) as "totalLikes"
+    from "likePosts" as "l"
+    where "postId" = $1
+    `;
+  const secondSql = `
+    select count("l".*) > 0 as "userLiked"
+    from "likePosts" as "l"
+    where "postId" = $1 and "userId" = $2
+  `;
+  const firstParams = [postId];
+  const secondParams = [postId, 1];
+
+  db.query(firstSql, firstParams)
+    .then(firstResult => {
+      if (!firstResult.rows) {
+        throw new ClientError(400, `cannot find post with postId ${postId}`);
+      }
+      return db.query(secondSql, secondParams)
+        .then(secondResult => {
+          if (!secondResult.rows) {
+            throw new ClientError(400, `cannot find post with postId ${postId}`);
+          }
+          const [totalLikes] = firstResult.rows;
+          const [userLiked] = secondResult.rows;
+          const likeData = { ...totalLikes, ...userLiked };
+          res.json(likeData);
+        })
+        .catch(err => next(err));
+    }).catch(err => next(err));
+});
+
+app.delete('/api/likes/:postId', (req, res, next) => {
+  const postId = Number(req.params.postId);
+  const sql = `
+    delete from "likePosts"
+    where "postId" = $1 and "userId" = $2
+    returning *
+  `;
+  const params = [postId, 1];
+
+  db.query(sql, params)
+    .then(result => {
+      if (!result.rows) {
+        throw new ClientError(400, `cannot find post with postId ${postId}`);
+      }
```

Comment:
This will never happen, since arrays are truthy (even empty arrays). You can leave this conditional out.